tests/src/ViewModel/LoginControlLoggedInTest.php

```diff
@@ -0,0 +1,217 @@
+            'linkFieldData' => 'data-profile-manage-uri="/profileManageURI" data-profile-manage-text="Manage my profile" data-logout-uri="/log-out" data-logout-text="Log out"',
+        ],
+    ];
+
+    /**
+     * @test
+     */
+    public function it_has_data()
+    {
+        $data = [
+            'displayName' => 'My name',
+            'isLoggedIn' => true,
+            'linkFields' => $this->linkFields['input'],
+            'defaultUri' => '/defaultUri',
+        ];
+
+        $profileLoginControl = LoginControl::loggedIn($data['defaultUri'], $data['displayName'], $this->linkFields['input']);
+
+        $this->assertSame($data['isLoggedIn'], $profileLoginControl['isLoggedIn']);
+        $this->assertSame($data['defaultUri'], $profileLoginControl['defaultUri']);
+
+        $this->assertSame($this->linkFields['expectedOutput']['linkFieldRoots'], $profileLoginControl['linkFieldRoots']);
+        $this->assertSame($this->linkFields['expectedOutput']['linkFieldData'], $profileLoginControl['linkFieldData']);
+
+        // Ugly hack to get the properties in the correct order for the following assertSame :-(
+        $data['linkFieldData'] = $this->linkFields['expectedOutput']['linkFieldData'];
+        $data['linkFieldRoots'] = $this->linkFields['expectedOutput']['linkFieldRoots'];
+        unset($data['linkFields']);
+        unset($data['defaultUri']);
+        $data['defaultUri'] = '/defaultUri';
+
+        $this->assertSame($data, $profileLoginControl->toArray());
+    }
+
+    /**
+     * @test
+     */
+    public function it_must_indicate_it_is_logged_in()
+    {
+        $profileLoginControl = LoginControl::loggedIn('/defaultUri', 'Display Name', $this->linkFields['input']);
+        $this->assertTrue($profileLoginControl['isLoggedIn']);
+    }
+
+    /**
+     * @test
+     */
+    public function it_must_be_given_a_profile_home_link()
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        LoginControl::loggedIn('', 'Display Name', $this->linkFields['input']);
+    }
+
+    /**
+     * @test
+     */
+    public function it_must_be_given_a_display_name()
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        LoginControl::loggedIn('/defaultUri', '', $this->linkFields['input']);
+    }
+
+    /**
+     * @test
+     */
+    public function it_must_be_given_link_fields()
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        LoginControl::loggedIn('/defaultUri', 'Display Name', []);
+    }
+
+    /**
+     * @test
+     */
+    public function it_must_have_a_text_data_attribute_for_each_data_attribute_root()
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        LoginControl::loggedIn(
+            '/defaultUri',
+            'Display Name',
+            [
+                'root-name-1' => [
+                    'uri' => '/uriOne',
+                    'text' => 'Text one',
```

Comment:
There's no need to require an ID like `root-name-1`. The input can just be `['Text one' => '/uriOne']`.